awx/main/scheduler/task_manager.py

```diff
@@ -66,17 +67,31 @@ def __init__(self):
         # ensures each task manager metric will be overridden when pipe_execute
         # is called later.
         self.subsystem_metrics = s_metrics.Metrics(auto_pipe_execute=False)
+        self.start_time = tz_now()
+        self.start_task_limit = settings.START_TASK_LIMIT
         for m in self.subsystem_metrics.METRICS:
             if m.startswith(self.prefix):
                 self.subsystem_metrics.set(m, 0)
 
-    def get_running_workflow_jobs(self):
-        graph_workflow_jobs = [wf for wf in WorkflowJob.objects.filter(status='running')]
-        return graph_workflow_jobs
+    def timed_out(self):
+        """Return True/False if we have met or exceeded the timeout for the task manager."""
+        elapsed = tz_now() - self.start_time
+        if elapsed.total_seconds() >= settings.TASK_MANAGER_TIMEOUT:
+            logger.warning(f"{self.prefix} manager has run for {elapsed} which is greater than TASK_MANAGER_TIMEOUT of {settings.TASK_MANAGER_TIMEOUT}.")
+            return True
+        return False
 
     @timeit
     def get_tasks(self, filter_args):
-        qs = UnifiedJob.objects.filter(**filter_args).exclude(launch_type='sync').order_by('created').prefetch_related('instance_group')
+        # We exclude workflows (both from WorkflowJobTemplates as well as sliced jobs)
+        workflow_ctype_id = ContentType.objects.get_for_model(WorkflowJob).id
+        qs = (
+            UnifiedJob.objects.filter(**filter_args)
+            .exclude(launch_type='sync')
+            .exclude(polymorphic_ctype=workflow_ctype_id)
+            .order_by('created')
+            .prefetch_related('instance_group')
+        )
         return [task for task in qs if not type(task) is WorkflowJob]
```

Comment:
is this conditional on WorkflowJob necessary if we do `.exclude(polymorphic_ctype=workflow_ctype_id)` above?